backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceQueueFileHelper.java

```diff
@@ -0,0 +1,112 @@
+package backtraceio.library.breadcrumbs;
+
+import com.squareup.tape.QueueFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+
+import backtraceio.library.logger.BacktraceLogger;
+
+public class BacktraceQueueFileHelper {
+    /**
+     * The base directory of the breadcrumb logs
+     */
+    private String breadcrumbLogDirectory;
+
+    /**
+     * The breadcrumb storage file
+     */
+    private QueueFile breadcrumbStore;
+
+    private final String LOG_TAG = BacktraceQueueFileHelper.class.getSimpleName();
+
+    private int maxQueueFileSizeBytes;
+
+    // This minimum file size comes from QueueFile::INITIAL_LENGTH
+    private final int minimumQueueFileSizeBytes = 4096;
+
+    private final Method usedBytes;
+
+    // Let our exceptions bubble all the way up to BacktraceBreadcrumbsLogger constructor
+    // We definitely cannot construct BacktraceBreadcrumbsLogger without an open file log
+    public BacktraceQueueFileHelper(String breadcrumbLogDirectory, int maxQueueFileSizeBytes) throws IOException, NoSuchMethodException {
+        this.breadcrumbLogDirectory = breadcrumbLogDirectory;
+        breadcrumbStore = new QueueFile(new File(this.breadcrumbLogDirectory));
```

Comment:
Here we're creating a breadcrumb file, but the directory, where the breadcrumb file exists, was created in the BacktraceBreadcrumbs class. Does it make sense to move directory creation code here as well to have it in one place? I think `BacktraceQueueFileHelper` is a better class to maintain file structure (including directory structure) than BacktraceBreadcrumbs.

Reply:
You bring up a good point. I will move the directory creation logic to `BacktraceBreadcrumbsLogger`.